### PR TITLE
[Feature] Add `modules.OrnsteinUhlenbeckProcessModule`

### PR DESCRIPTION
**Pull request by kurtamohler**
## Description

Add `OrnsteinUhlenbeckProcessModule`, which serves the same purpose as `OrnsteinUhlenbeckProcessWrapper`, but derives from `TensorDictModuleBase` rather than `TensorDictModuleWrapper`

## Motivation and Context

part of #2295

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] New feature (non-breaking change which adds core functionality)
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
LGTM
nit: Since we want to deprecate the OU wrapper, can you add a deprecation warning? 2 releases from 0.5, so it will be in 0.7!
Thanks!

File: docs/source/reference/modules.rst [hunk 1]
```diff
@@ -75,6 +75,7 @@ other cases, the action written in the tensordict is simply the network output.
     AdditiveGaussianWrapper
     EGreedyModule
     EGreedyWrapper
+    OrnsteinUhlenbeckProcessModule
     OrnsteinUhlenbeckProcessWrapper
```
Comment (vmoens) [inline docs/source/reference/modules.rst:79]:
```suggestion
    OrnsteinUhlenbeckProcessModule
```
since we want to deprecate it

Comment (vmoens) [reply docs/source/reference/modules.rst:79]: hmm upon reflection, let's keep this one, people may still see it in their code

**Comment by kurtamohler** [comment]
I've added the deprecation warning

File: docs/source/reference/modules.rst (hunk 1 above)
Comment (kurtamohler) [reply docs/source/reference/modules.rst:79]: Yeah I'd think it's typically a good idea to keep the old documentation and only remove it at the point when the deprecated feature is actually removed

**Pull request merged**